tests/test_utils.py

```diff
@@ -89,10 +86,63 @@ def make_links_blue(node):
                 "children": ["universe"],
                 "attributes": {"style": {"color": "blue"}},
             },
+            "lmao",
+        ],
+    }
+
+    assert html_to_vdom(source, make_links_blue) == expected
+
+
+def test_non_html_tag_behavior():
+    source = "<my-tag data-x=something><my-other-tag key=a-key /></my-tag>"
```

Comment:
It seems like `lxml` doesn't allow custom tags like this even though, if you define something like a web component, tags like this are valid. Maybe we should use `bs4`?

Reply:
It would allow custom tags like this if we disable strict parsing.

Reply:
It seems like a lot to give up the benefit of the strict parsing just to get custom tags. Does `bs4` have a strict mode that behaves similarly? If not, then maybe we expose that strict flag after all.

Reply:
I don't recall ever seeing a toggleable option for strictness of the BS4 parser, I always assumed it was permanently lax.

I'll see if I can find any info in the docs.

Reply:
we just added the strict flag + descriptive error message